### ISSUE TITLE
About Visualization

### ISSUE THREAD

**Issue by CherryDLAM** [kosukesakurai1/VLP-SAM#3]
Hello! I hope to perform some visualization operations on this code, and in the future, I also plan to replace the dataset. Would you be able to provide me with a few suggestions? I truly appreciate your help and look forward to hearing from you soon.

**Comment by kosukesakurai1** [comment]
Thank you for your interest in this project.

We regret to inform you that we are currently in the process of submitting a paper, and thus we are unable to share additional code. For visualization, we recommend referring to the 'VLP-SAM/common/vis.py'.

**Comment by bowen099** [comment]
I'm currently visualizing the results, but due to my lack of experience, I've run into a problem.
The predicted mask seems fine, but the issue appears when visualizing the support image together with the ground truth mask — they don't align properly.
Do you have any advice or experience to share on this?
Sorry to bother you, and thank you again!

![Image](https://github.com/user-attachments/assets/9a1b4802-c09b-44f2-baa5-ec6b5bfcacac)

**Comment by foreverzzx** [comment]
> I'm currently visualizing the results, but due to my lack of experience, I've run into a problem. The predicted mask seems fine, but the issue appears when visualizing the support image together with the ground truth mask — they don't align properly. Do you have any advice or experience to share on this? Sorry to bother you, and thank you again!
> 
> ![Image](https://github.com/user-attachments/assets/9a1b4802-c09b-44f2-baa5-ec6b5bfcacac)

I face the same problem, have you solve it?

**Comment by foreverzzx** [comment]
I resolved the issue. The inconsistency between the support images and their corresponding masks was due to different resizing methods applied in the dataset pipeline.
In clip.py, the support images are resized using the following operation:
`Resize(n_px, interpolation=BICUBIC)`
This maintains the aspect ratio of the image.

However, in coco.py, the corresponding masks were resized using:
`support_masks[midx] = F.interpolate(smask.unsqueeze(0).unsqueeze(0).float(), support_imgs.size()[-2:], mode='nearest').squeeze()`
This approach does not preserve the original aspect ratio—both the height and width are forcibly set to match the target dimensions.

To fix this mismatch, I modified the mask resizing logic as follows:
`
import torchvision.transforms.functional as TF
support_masks[midx] = TF.resize(smask.unsqueeze(0), support_imgs.size()[-1], interpolation=TF.InterpolationMode.NEAREST)
support_masks[midx] = TF.center_crop(support_masks[midx], support_imgs.size()[-1]).squeeze()
`
This ensures the mask resizing now aligns with the image resizing method, preserving consistency.

Enjoy!

**Comment by bowen099** [comment]
Thank you for your answer. I have successfully fixed it.
I feel that this issue might be because this project integrates CLIP but overlooks the consistency of image resizing, whereas other FSS methods don't use CLIP, so they don't have this problem.

**Comment by foreverzzx** [comment]
> Thank you for your answer. I have successfully fixed it. I feel that this issue might be because this project integrates CLIP but overlooks the consistency of image resizing, whereas other FSS methods don't use CLIP, so they don't have this problem.

By the way, do you know any other great project about visual prompt for segmentation? Can you share them? Thanks.

**Comment by CherryDLAM** [comment]
About zero-shot Segmentation: ZegCLIP , ESDA: Zero-Shot Semantic Segmentation Based on an Embedding Semantic Space Distribution Adjustment Strategy
About few-shot:Learnable Prompt for Few-Shot Semantic Segmentation

> > 谢谢你的回答。我已经成功解决了。我觉得这个问题可能是因为这个项目集成了 CLIP，但忽略了图像缩放的一致性，而其他 FSS 方法没有使用 CLIP，所以不存在这个问题。
> 
> 顺便问一下，您还知道其他关于视觉提示分割的优秀项目吗？可以分享一下吗？谢谢。

**Comment by CherryDLAM** [comment]
> 问题已解决。支持图像与其对应掩码之间的不一致是由于数据集处理流程中采用了不同的图像缩放方法造成的。 在 clip.py 文件中，支持图像使用以下操作进行缩放： `Resize(n_px, interpolation=BICUBIC)` 这样可以保持图像的宽高比。
> 
> 然而，在 coco.py 中，相应的掩码是使用以下方式调整大小的： `support_masks[midx] = F.interpolate(smask.unsqueeze(0).unsqueeze(0).float(), support_imgs.size()[-2:], mode='nearest').squeeze()` 这种方法不会保留原始的纵横比——高度和宽度都被强制设置为与目标尺寸匹配。
> 
> 为了解决这种不匹配问题，我修改了掩码调整大小的逻辑如下： `import torchvision.transforms.functional as TF support_masks[midx] = TF.resize(smask.unsqueeze(0), support_imgs.size()[-1], interpolation=TF.InterpolationMode.NEAREST) support_masks[midx] = TF.center_crop(support_masks[midx], support_imgs.size()[-1]).squeeze()` 这样可以确保掩码调整大小与图像调整大小的方法保持一致，从而保持一致性。
> 
> 享受！

Due to the availability of only a single GPU, I modified the training setup by switching from distributed training to single-GPU training. However, the final mean Intersection over Union (mIoU) score reached only 16. Have you encountered similar performance degradation under such conditions? I would appreciate your insights on this matter.
